File: rust/src/water_bottles_1518.rs
use crate::Solution;
impl Solution {
    pub fn num_water_bottles(num_bottles: i32, num_exchange: i32) -> i32 {
        if num_exchange > num_bottles {
            return num_bottles;
        } else {
            return num_bottles + (num_bottles - 1) / (num_exchange - 1);
        }
    }
}

File: python/water_bottles_1518.py
class Solution:
    def numWaterBottles(self, numBottles, numExchange):
        #    total_drunk = 0
        #    empty_bottles = 0
        #
        #    while numBottles > 0:
        #        total_drunk += numBottles
        #        empty_bottles += numBottles
        #        numBottles = empty_bottles // numExchange
        #        empty_bottles = empty_bottles % numExchange
        #    return total_drunk
        #
        if numExchange > numBottles:
            return numBottles
        else:
            return numBottles + (numBottles - 1) // (numExchange - 1)
